pkg/kwok/controllers/stage_controller.go

```diff
@@ -310,32 +315,28 @@ func (c *StageController) playStage(ctx context.Context, resource *unstructured.
 	if next.Finalizers != nil {
 		result, err = c.finalizersModify(ctx, resource, next.Finalizers)
 		if err != nil {
-			logger.Error("Failed to finalizers", err)
-			return
+			return shouldRetry(err), fmt.Errorf("failed to patch the finalizer of resource %s: %w", resource.GetName(), err)
 		}
 	}
 
 	if next.Delete {
 		err = c.deleteResource(ctx, resource)
 		if err != nil {
-			logger.Error("Failed to delete resource", err)
-			return
+			return shouldRetry(err), fmt.Errorf("failed to delete resource %s: %w", resource.GetName(), err)
 		}
 	} else if next.StatusTemplate != "" {
-		patch, err = c.configureResource(resource, next.StatusTemplate)
+		patch, err = c.computeStatusPatch(resource, next.StatusTemplate)
 		if err != nil {
-			logger.Error("Failed to configure resource", err)
-			return
+			return shouldRetry(err), fmt.Errorf("failed to compute the status patch of resource %s: %w", resource.GetName(), err)
 		}
 		if patch == nil {
 			logger.Debug("Skip resource",
 				"reason", "do not need to modify",
 			)
 		} else {
-			result, err = c.patchResource(ctx, resource, patch)
+			result, err = c.patchResource(ctx, resource, types.StrategicMergePatchType, patch, "status")
```

Comment:
```suggestion
			result, err = c.patchResource(ctx, resource, types.MergePatchType, patch, "status")
```

This is not the same as Pod and Node, using `types.StrategicMergePatchType` will cause patch failure in some cases, but I forget the specific reason.